File: inkatravel-backend/src/main/java/com/inkatravel/service/ReservaService.java
package com.inkatravel.service;

import com.inkatravel.dto.ReservaRequestDTO;
import com.inkatravel.dto.ReservaResponseDTO;
import java.security.Principal;
import java.util.List; // <-- NUEVA IMPORTACIÓN

public interface ReservaService {

    /**
     * (RF-08) Crea una nueva reserva (PENDIENTE).
     */
    ReservaResponseDTO crearReserva(ReservaRequestDTO reservaDTO, Principal principal) throws Exception;

    /**
     * (NUEVO - RF-11) Obtiene el historial de reservas del usuario autenticado.
     * @param principal El usuario autenticado.
     * @return Una lista de sus reservas (en formato DTO).
     */
    List<ReservaResponseDTO> obtenerMisReservas(Principal principal) throws Exception;
}

File: inkatravel-frontend/src/app/features/reservas/services/reserva.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { ReservaResponseDTO, ReservaRequestDTO } from '../../../core/interfaces/reserva.interface';
import { environment } from '../../../../environments/environment';


@Injectable({
  providedIn: 'root'
})
export class ReservaService {

  private baseUrl = `${environment.apiUrl}/reservas`;
  private http = inject(HttpClient);

  constructor() { }

  /**
   * (RF-08) Crea una nueva reserva PENDIENTE en el backend.
   * Llama a: POST /api/reservas (protegido por JWT)
   */
  crearReserva(reservaData: ReservaRequestDTO): Observable<ReservaResponseDTO> {
    return this.http.post<ReservaResponseDTO>(this.baseUrl, reservaData);
  }

  /**
   * (RF-11) Obtiene las reservas del usuario logueado.
   * Llama a: GET /api/reservas/mis-reservas (protegido por JWT)
   */
  obtenerMisReservas(): Observable<ReservaResponseDTO[]> {
    return this.http.get<ReservaResponseDTO[]>(`${this.baseUrl}/mis-reservas`);
  }
}